Poor compilation warning for error propogation in loop

### Describe the bug

When using an error propagation in a for loop, I get an error message: 

```
>>> v -cg .
/tmp/v_1001/vsweeper.161389303863868986.tmp.c:12580: error: ')' expected (got "_t1")
```

This is very unhelpful in debugging and finding the error, and confusing to newcomers to vlang like me. Ideally, we'd get something similar to when trying to use the alternative `or` for options.

```
src/main.v:9:24: error: options are not allowed in `for statement increment` (yet)
    7 | fn main() {
    8 |     mut pos := 0
    9 |     for ; pos > 10; pos = foo() or { 6 } {
      |                           ~~~~~
   10 |         println(pos)
   11 |     }
```

### Reproduction Steps

Run each each of the loops below with `v -cg .`

```vlang
module main

fn foo() !int {
	return 5
}

fn main() {
	mut pos := 0

	// /tmp/v_1001/vsweeper.161389303863868986.tmp.c:12580: error: ')' expected (got "_t1")
	for ; pos > 10; pos = foo()! {
		println(pos)
	}

	// /tmp/v_1001/vsweeper.7081838145059576425.tmp.c:12579: error: identifier expected
	for pos = foo()!; pos > 10; {
		println(pos)
	}

	// no error
	// for ; pos < foo()!; {
	// 	println(pos)
	// }
}
```



### Expected Behavior

Output the same or similar `options are not allowed in 'for statement increment' (yet)` as when using error propagation in for loop.

### Current Behavior

Compilation fails with warnings:

```
/tmp/v_1001/vsweeper.161389303863868986.tmp.c:12580: error: ')' expected (got "_t1")
```

and

```
/tmp/v_1001/vsweeper.7081838145059576425.tmp.c:12579: error: identifier expected
```

### Possible Solution

Might be possible to handle this the same way `v` checks for the `for i :=0; i < 10; i = foo! { or 5 }`

### Additional Information/Context

_No response_

### V version

V 0.4.2 65d7126

### Environment details (OS name and version, etc.)

V full version: V 0.4.2 7e2d737.65d7126
OS: linux, Ubuntu 22.04.3 LTS
Processor: 16 cpus, 64bit, little endian, 13th Gen Intel(R) Core(TM) i7-1360P

getwd: /home/wrinkle/workspaces/joshmeranda/vsweeper
vexe: /home/wrinkle/workspaces/vlang/v/v
vexe mtime: 2023-10-23 01:52:31

vroot: OK, value: /home/wrinkle/workspaces/vlang/v
VMODULES: OK, value: /home/wrinkle/.vmodules
VTMP: OK, value: /tmp/v_1001

Git version: git version 2.34.1
Git vroot status: weekly.2023.42-28-g65d71269
.git/config present: true

CC version: cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/for.v
+++ b/vlib/v/checker/for.v
@@ -19,15 +19,6 @@ fn (mut c Checker) for_c_stmt(mut node ast.ForCStmt) {
 			if assign.op == .decl_assign {
 				c.error('for loop post statement cannot be a variable declaration', assign.pos)
 			}
-
-			for right in assign.right {
-				if right is ast.CallExpr {
-					if right.or_block.stmts.len > 0 {
-						c.error('options are not allowed in `for statement increment` (yet)',
-							right.pos)
-					}
-				}
-			}
 		}
 		c.stmt(mut node.inc)
 	}

--- a/vlib/v/checker/tests/for_c_stmt_with_option_call.out
+++ b/vlib/v/checker/tests/for_c_stmt_with_option_call.out
@@ -1,7 +1,0 @@
-vlib/v/checker/tests/for_c_stmt_with_option_call.vv:15:31: error: options are not allowed in `for statement increment` (yet)
-   13 |     for t := 0; t < tests; t++ {
-   14 |         mut v := []bool{len: nmax, init: false}
-   15 |         for x := 0; !v[x]; x = rand.intn(n) or { 0 } {
-      |                                     ~~~~~~~
-   16 |             v[x] = true
-   17 |             sum++

--- a/vlib/v/gen/c/for.v
+++ b/vlib/v/gen/c/for.v
@@ -56,6 +56,8 @@ fn (mut g Gen) for_c_stmt(node ast.ForCStmt) {
 		if node.label.len > 0 {
 			g.writeln('${node.label}:')
 		}
+		g.set_current_pos_as_last_stmt_pos()
+		g.skip_stmt_pos = true
 		g.write('for (')
 		if !node.has_init {
 			g.write('; ')
@@ -93,6 +95,7 @@ fn (mut g Gen) for_c_stmt(node ast.ForCStmt) {
 			}
 		}
 		g.writeln(') {')
+		g.skip_stmt_pos = false
 		g.is_vlines_enabled = true
 		g.inside_for_c_stmt = false
 		g.stmts(node.stmts)

--- a/vlib/v/tests/for_c_stmt_with_option_call_test.v
+++ b/vlib/v/tests/for_c_stmt_with_option_call_test.v
@@ -20,7 +20,7 @@ fn avg(n int) f64 {
 	return f64(sum) / tests
 }
 
-fn main() {
+fn test_for_c_stmt_with_option_call() {
 	println(' N   average   analytical   (error)')
 	println('=== ========= ============ =========')
 	for n in 1 .. nmax + 1 {
@@ -28,4 +28,5 @@ fn main() {
 		b := ana(n)
 		println('${n:3} ${a:9.4f} ${b:12.4f} (${(abs(a - b) / b * 100):6.2f}%)')
 	}
+	assert true
 }

--- a/vlib/v/tests/for_c_stmt_with_result_call_test.v
+++ b/vlib/v/tests/for_c_stmt_with_result_call_test.v
@@ -1,0 +1,19 @@
+module main
+
+fn foo() !int {
+	return 5
+}
+
+fn test_for_c_stmt_with_result_call() {
+	mut pos := 0
+
+	for ; pos > 10; pos = foo()! {
+		println(pos)
+	}
+
+	for pos = foo()!; pos > 10; {
+		println(pos)
+	}
+
+	assert true
+}

